File: 04592.cpp
#include <iostream>
using namespace std;

int main() {
	while (1) {
		int N; cin >> N;
		if (!N) break;

		int num = -1;
		for (int i = 0; i < N; i++) {
			int n; cin >> n;

			if (n != num) {
				num = n;
				cout << num << ' ';
			}
		}

		cout << "$\n";
	}
}

File: 04592.py
while 1:
    N = list(map(int, input().split()))
    if N[0] == 0: break

    num = -1
    for i in range(1, len(N)):
        if N[i] != num:
            num = N[i]
            print(num, end=' ')

    print('$')
